tests/utils/__init__.py

```diff
@@ -79,26 +79,38 @@ def get_generate_root_otp():
     return test_otp
 
 
-def create_client(url="https://localhost:8200", **kwargs):
+def create_client(url="https://localhost:8200", use_env=False, **kwargs):
     """Small helper to instantiate a :py:class:`hvac.v1.Client` class with the appropriate parameters for the test env.
 
     :param url: Vault address to configure the client with.
     :type url: str
+    :param use_env: configure vault using environment variable
+    :type use_env: bool
     :param kwargs: Dictionary of additional keyword arguments to pass through to the Client instance being created.
     :type kwargs: dict
     :return: Instantiated :py:class:`hvac.v1.Client` class.
     :rtype: hvac.v1.Client
     """
+
     client_cert_path = get_config_file_path("client-cert.pem")
     client_key_path = get_config_file_path("client-key.pem")
     server_cert_path = get_config_file_path("server-cert.pem")
-
-    return Client(
-        url=url,
-        cert=(client_cert_path, client_key_path),
-        verify=server_cert_path,
-        **kwargs,
-    )
+    if use_env:
+        with mock.patch("hvac.v1.VAULT_CAPATH", server_cert_path):
+            with mock.patch("hvac.v1.VAULT_CLIENT_CERT", client_cert_path):
+                with mock.patch("hvac.v1.VAULT_CLIENT_KEY", client_key_path):
```

Comment:
I might consider using `with mock.patch.dict` here against `os.env`, that way you can provide a dict of env vars to override.

Reply:
@briantist i don’t think that will work since the vault environment variables are read at import time of the constants module before this test case patch would be applied. Also if it did work it would end up leaving the global variables in that state after the test completes.

Reply:
Thanks that  makes sense. Opened https://github.com/hvac/hvac/issues/945 to consider whether this should change in the future.